pkg/core/dao/dao.go

```diff
@@ -180,7 +177,7 @@ func (dao *Simple) putContractScriptHash(cs *state.Contract) error {
 	key := make([]byte, 5)
 	key[0] = byte(storage.STContractID)
 	binary.LittleEndian.PutUint32(key[1:], uint32(cs.ID))
-	return dao.Store.Put(key, cs.ScriptHash().BytesBE())
+	return dao.Store.Put(key, cs.Hash.BytesBE())
```

Comment:
We may perform a tiny optimisation here. At the current moment `putContractScriptHash` is called each time when we create or update the contract. But as far as the hash is constant now, we can add `isUpdate` flag to`PutContractState` and do not call `putContractScriptHash` during contract's update.